app/views/shared/_navbar.html.haml

```diff
@@ -1,15 +1,21 @@
 %nav.navbar.navbar-static-top{ role: "navigation" }
   .navbar-custom-menu
     %ul.nav.navbar-nav
-      %li
-        = link_to setting_path do
-          %i.fa.fa-gear
-          Setting
-      %li
-        = link_to edit_user_path(current_user) do
-          %i.fa.fa-user
-          = current_user.name
-      %li
-        = link_to logout_path, method: :delete do
-          %i.fa.fa-sign-out
-          Sign-out
+      - if current_user
+        %li
+          = link_to setting_path do
+            %i.fa.fa-gear
+            Setting
+        %li
+          = link_to edit_user_path(current_user) do
+            %i.fa.fa-user
+            = current_user.name
+        %li
+          = link_to logout_path, method: :delete do
+            %i.fa.fa-sign-out
+            Sign-out
+      - else
+        %li
+          = link_to google_oauth2_path(state: url_for(params.merge(only_path: true))) do
+            %i.fa.fa-sign-in
+            Login
```

Comment:
I prefer `Sign-in` here.

dmemo using `Sign-out` yet https://github.com/hogelog/dmemo/pull/93/files#diff-fdf07dae26df4b84203e7a8380c4fdcaR16